EasyPost/Hooks.cs

```diff
@@ -0,0 +1,135 @@
+    /// </summary>
+    public Guid Id { get; }
+
+    /// <summary>
+    ///     Constructs a new instance of the <see cref="OnRequestExecutingEventArgs"/> class.
+    /// </summary>
+    /// <param name="request">The <see cref="HttpRequestMessage"/> about to be executed by the HTTP request.</param>
+    /// <param name="timestamp">The timestamp of the HTTP request.</param>
+    /// <param name="guid">A unique identifier for the HTTP request-response pair.</param>
+    internal OnRequestExecutingEventArgs(HttpRequestMessage request, int timestamp, Guid guid)
+    {
+        Request = request;
+        RequestTimestamp = timestamp;
+        Id = guid;
+    }
+}
+
+public class OnRequestResponseReceivedEventArgs : EventArgs
+{
+    /// <summary>
+    ///     The <see cref="HttpResponseMessage"/> returned by the HTTP request.
+    /// </summary>
+    private HttpResponseMessage Response { get; } // Not publicly-exposed, used for reference only.
+
+    /// <summary>
+    ///     The <see cref="HttpStatusCode"/> of the HTTP response.
+    /// </summary>
+    public HttpStatusCode ResponseStatusCode => Response.StatusCode;
+
+    /// <summary>
+    ///     The <see cref="HttpMethod"/> of the HTTP request that prompted the associated response.
+    /// </summary>
+    public HttpMethod? RequestMethod => Response.RequestMessage?.Method;
+
+    /// <summary>
+    ///     The <see cref="Uri"/> of the HTTP request that prompted the associated response.
+    /// </summary>
+    public Uri? RequestUri => Response.RequestMessage?.RequestUri;
+
+    /// <summary>
+    ///     The <see cref="HttpHeaders"/> present in the HTTP response.
+    /// </summary>
+    public HttpResponseHeaders ResponseHeaders => Response.Headers;
+
+    /// <summary>
+    ///     The <see cref="HttpContent"/> of the HTTP response.
+    /// </summary>
+    public HttpContent ResponseBody => Response.Content;
+
+    /// <summary>
+    ///     The timestamp of the HTTP request.
+    /// </summary>
+    public int RequestTimestamp { get; }
+
+    /// <summary>
+    ///     The timestamp of the HTTP response.
+    /// </summary>
+    public int ResponseTimestamp { get; }
+
+    /// <summary>
+    ///     A unique identifier for the HTTP request-response pair.
+    /// </summary>
+    public Guid Id { get; }
+
+    /// <summary>
+    ///     Constructs a new instance of the <see cref="OnRequestResponseReceivedEventArgs"/> class.
+    /// </summary>
+    /// <param name="response">The <see cref="HttpResponseMessage"/> returned by the HTTP request.</param>
+    /// <param name="requestTimestamp">The timestamp of the HTTP request.</param>
+    /// <param name="responseTimestamp">The timestamp of the HTTP response.</param>
+    /// <param name="guid">A unique identifier for the HTTP request-response pair.</param>
+    internal OnRequestResponseReceivedEventArgs(HttpResponseMessage response, int requestTimestamp, int responseTimestamp, Guid guid)
```

Comment:
Why does the constructor only have four parameters?

Reply:
Object consumes an `HttpResponseMessage` object and stores it privately, all expected properties are derived from it internally.